src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/4.diff.sql

```diff
@@ -0,0 +1,78 @@
+/*************************************************************
+    Search Parameter Status Information
+**************************************************************/
+
+ALTER TABLE dbo.SearchParam
+ADD
+    Status varchar(10) NOT NULL,
+    LastUpdated datetimeoffset(7) NOT NULL,
+    IsPartiallySupported bit NOT NULL
+
+CREATE TYPE dbo.SearchParamTableType_1 AS TABLE
+(
+    Uri varchar(128) COLLATE Latin1_General_100_CS_AS NOT NULL,
+    Status varchar(10) NOT NULL,
+    IsPartiallySupported bit NOT NULL
+)
+
+GO
+
+/*************************************************************
+    Stored procedures for search parameter information
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     Gets all the search parameters and their statuses.
+--
+-- DESCRIPTION
+--     Retrieves and returns select contents from the search parameter table.
+--
+-- RETURN VALUE
+--     The search parameters and their statuses.
+--
+CREATE PROCEDURE dbo.GetSearchParamStatuses
+AS
+    SET NOCOUNT ON
+
+    SELECT Uri, Status, LastUpdated, IsPartiallySupported FROM dbo.SearchParam
+GO
+
+--
+-- STORED PROCEDURE
+--     Given a set of search parameters, creates or updates the parameters.
+--
+-- DESCRIPTION
+--     If a parameter with a matching URI already exists in the table, it is updated.
+--     If not, a new entry is created.
+--
+-- PARAMETERS
+--     @searchParams
+--         * The updated or new search parameters
+--
+CREATE PROCEDURE dbo.UpsertSearchParams
+    @searchParams dbo.SearchParamTableType_1 READONLY
+AS
+    SET NOCOUNT ON
+    SET XACT_ABORT ON
+
+    SET TRANSACTION ISOLATION LEVEL SERIALIZABLE
+    BEGIN TRANSACTION
+
+    DECLARE @lastUpdated datetimeoffset(7) = SYSDATETIMEOFFSET()
+
+    -- Acquire and hold an exclusive table lock for the entire transaction to prevent parameters from being added or modified during upsertion.
```

Comment:
nit: upsertion -> upsert

Reply:
Haha I think I made up the word "upsertion"... thank you for catching it. Fixed in 0bae8e3.